controllers/routing/routing_controller_test.go

```diff
@@ -120,6 +117,9 @@ var _ = Describe("Platform routing setup for the component", test.EnvTest(), fun
 
 			// when routing through platform mesh
 			By("adding component service to platform routing", func() {
+				// before updating component, ensure finalizers have been set
+				component = ensureFinalizersSet(ctx, component)
```

Comment:
Maybe instead of waiting till we handle `CREATE` we could  call `exportCustomResource(ctx, component, "external")` and create once instead? That would suffice and reflect the real scenario, as we expect the component to have those annotations from the get-go.

So the whole structure of each test would be more like

```
// when
creating exported component and service

// expect 
routing resources checks
```

Reply:
[6ab163f](https://github.com/maistra/odh-platform/pull/39/commits/6ab163f65fdc21bdfaace859171e2f1c5492383a)

agreed, less `update` calls now